packages/venia-ui/lib/util/shrinkGETQuery.js

```diff
@@ -0,0 +1,32 @@
+import { stripIgnoredCharacters } from 'graphql/utilities/stripIgnoredCharacters';
+
+/**
+ * Shrink a GraphQL query inside of a URL used in a GET request.
+ * There are 2 problems with Apollo-client's encoding of URLs:
```

Comment:
Did you find any issues or documentation about this? It would be good to link them here for posterity if so - it'd be cool if we didn't have to do this ourselves.

Reply:
Planning on opening a PR to `apollo-client` to discuss the changes with them. Haven't quite gotten there yet, though!

Reply:
There are some tools today that already use the `stripIgnoreCharacters` utility for this (like [`babel-plugin-graphql-tag`](https://github.com/gajus/babel-plugin-graphql-tag/blob/e76450c2cf667ebdac85dc34a0dc5d3fce64ec8c/src/index.js#L10)), but I haven't seen any solutions that work for apollo client (since it doesn't use the string compiled by `babel-plugin-graphql-tag` or `graphql-tag/loader`)

And re: the querystring encoding, I haven't managed to find any prior discussions about it on the interwebs. Surprising though cause I'd doubt we're the first to notice this.

Reply:
> There are some tools today that already use the stripIgnoreCharacters utility for this (like babel-plugin-graphql-tag), but I haven't seen any solutions that work for apollo client (since it doesn't use the string compiled by babel-plugin-graphql-tag or graphql-tag/loader)

Huh, we use the babel plugin. It might shave some bytes on the bundle if we can do the optimization there rather than right before the fetch.

Reply:
> It might shave some bytes on the bundle if we can do the optimization there rather than right before the fetch.

This was actually the first version of my implementation. It turns out, apollo-client can modify the AST of the query for certain things, so they always serialize from the AST, rather than using the string generated at build time.

https://github.com/apollographql/apollo-client/blob/7993a778ef5a9e61bfee72b956f4c179fce691f4/src/link/http/selectHttpOptionsAndBody.ts#L132

So my original implementation saved a tiny bit on the size of queries in the bundle, but it raised a bigger issue: the bundle has _both_ the string query and the AST, but only the AST is used.

Reply:
Sounds like we should still take advantage of it to not have to include the 16kb stripIgnored utility, right? Or am I missing something :P

Reply:
> Sounds like we should still take advantage of it to not have to include the 16kb stripIgnored utility, right? Or am I missing something :P

Still needed. Apollo serializes the AST itself, and that serialization includes spaces/punctuators/etc. The query you write in your editor never actually goes over the wire - it just gets converted to an AST and inlined.